sympy/calculus/util.py

```diff
@@ -1523,7 +1523,7 @@ def _eval_is_ge(lhs, rhs): # noqa:F811
     if is_lt(lhs.max, rhs.min):
         return False
 
-@dispatch(AccumulationBounds, Expr)
+@dispatch(AccumulationBounds, Expr)  # type:ignore
```

Comment:
Maybe we should modify the `@dispatch` decorator in order to work with _mypy_.

Unfortunately decorators are unable to retrieve comments, so this won't work:
```python
@dispatch
def _eval_is_ge(lhs, rhs):  # type(AccumulationBounds, Expr)
```

but maybe this alternative syntax could work:
```python
@dispatch
def _eval_is_ge(lhs: AccumulationBounds, rhs: Expr):
```

There's a _mypy_ issue on repeated namings:
https://github.com/python/mypy/issues/2904

Reply:
There's also an issue with flake8 #17795 and https://github.com/PyCQA/pyflakes/issues/488

We can get round flake8 with this:
```python
from sympy.multipledispatch import Dispatcher

sub = Dispatcher('sub')

@sub.register(int, float)
def _(a, b):
    return a - b

@sub.register(tuple, list)
def _(a, b):
    return a - b
```
However that doesn't work with `mypy`.

Reply:
> but maybe this alternative syntax could work:

Neither of those would work. The issue is about repeatedly using the same function name. With `flake8` you can use `_` as the name to work around it but that doesn't work with `mypy`.

Reply:
What about just renaming the function (instead of waiting for the issue to be fixed)? `@sub.register( ... )` should ignore the function name.

Reply:
That would work but the question is do we want to do that just because of an issue in mypy or is it actually something that we want to do?

I'm not going to do it in this PR anyway as there are loads of these:
```
$ git grep dispatch | grep ignore | wc -l
      98
```
The ones that you see here are just those that have emerged in new code since #18244

Reply:
Yeah, maybe it's better to just ignore the current mypy bug. Though I really hope that some future version of Python will natively support multiple-dispatching.